--- emscripten_toolchain/emcc_link.bat
+++ emscripten_toolchain/emcc_link.bat
@@ -5,1 +5,11 @@
-py -3 %~dp0\link_wrapper.py %*
+set PYTHON_EXE=python
+if exist "%~dp0\..\..\rules_python++python+python_3_11_x86_64-pc-windows-msvc\python.exe" (
+    set PYTHON_EXE="%~dp0\..\..\rules_python++python+python_3_11_x86_64-pc-windows-msvc\python.exe"
+) else (
+    where.exe /q python
+    if %ERRORLEVEL% equ 0 set PYTHON_EXE=python
+    where.exe /q py
+    if %ERRORLEVEL% equ 0 set PYTHON_EXE=py -3
+)
+
+%PYTHON_EXE% %~dp0\link_wrapper.py %*
